lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -714,12 +714,14 @@ JavaScriptCompiler.prototype = {
 
         this.useDepths = this.useDepths || compiler.useDepths;
         this.useBlockParams = this.useBlockParams || compiler.useBlockParams;
+        child.useDepths = this.useDepths || compiler.useDepths;
+        child.useBlockParams = this.useBlockParams || compiler.useBlockParams;
```

Comment:
We don't need the `||` conditions here, it's already been done in the lines above.